doc/common-issues-and-pitfalls.rst

```diff
@@ -420,12 +420,12 @@ also interfere with the cleanup itself due to attempts at accessing the files.
 
 Therefore, if you need to spawn a subprocess using :data:`sys.executable`
 that will outlive the current application process, you need to ensure
-that is spawned as an independent instance. This is done by by clearing
-the ``_PYI_ARCHIVE_FILE`` environment variable before spawning the
-process, for example::
+that is spawned as an independent instance. This is done by setting
+the ``PYINSTALLER_RESET_ENVIRONMENT`` environment variable to ``1``
+before spawning the process, for example::
 
     # Restart the application
-    del os.environ['_PYI_ARCHIVE_FILE']
+    os.environ['PYINSTALLER_RESET_ENVIRONMENT'] = '1'
     subprocess.Popen([sys.executable])
```

Comment:
Presumably the current process doesn't need the variable so this would also do?
```python
subprocess.Popen([sys.executable], env={**os.environ, "PYINSTALLER_RESET_ENVIRONMENT": "1"))
```

Reply:
Indeed; the test script for `test_subprocess_environment_inheritance` also passes `env` to `subprocess.Popen` as opposed to modifying the caller's environment.